Jimena/docs updates part 2 (#4285)
SwissDataScienceCenter/renku: docs/docs/10-users/60-sessions/guides/98-create-streamlit-app.md

# How to create a Streamlit app on top of your project

:::info 

> Renku 2.0 will allow Docker images with user-specific entry points soon. This feature  will make it easier the creation of user-specific environments, including setting up Streamlit apps. 

Reviewer: ```suggestion
This documentation is out-of-date and will be updated soon!
```

:::

> As of now, it is still possible to launch your Streamlit app on top of your Renku 2.0 project as follows.

Reviewer: I do not follow the "As of now, it is still possible to..." -- as far as I know, we are not planning on removing this functionality.

```suggestion
It is possible to include your Streamlit app on in your Renku 2.0 project as follows.
```

1. Create a fork of this repository: https://gitlab.renkulab.io/rok.roskar/streamlit-v2-example, we will refer to the fork as project-fork.

<p class="image-container-l">
![image.png](./streamlit-v2-example.png)
</p>

2. In *project-fork*, modify line 8 in the file `jupyter_noteboox_config.py`to the path to your python entry file for Streamlit within your Renku 2.0 project ( e.g. `work/name-fork-project/helloworld.py` or  `work/main-project-repository/streamlit/app.py` or  `work/repository-2/app.py` )
    
    ⚠️ Do not forget to add `work/` at the beginning of the path !

<p class="image-container-l">
![image.png](./jupiter-notebook.png)
</p>

> 3. Add all the package dependencies you will need to run your Renku 2.0 project in the requirements.txt file or in the environment.yml , as explained [here](/docs/users/sessions/guides/install-packages-on-the-fly-in-your-session).

Reviewer: ```suggestion
3. Make sure all the dependencies for your Streamlit app are in the `requirements.txt` file or `environment.yml` , as explained [here](/docs/users/sessions/guides/install-packages-on-the-fly-in-your-session).
```
